.github/workflows/go-mod-auto-bump.yml

```diff
@@ -0,0 +1,37 @@
+# This workflow is used in order to automatically update go.mod in case one of:
+# * osmoutils
+# * osmomath
+# * ibc-hooks
+# is updated in local branch.
+name: Go Mod Auto Version Update
+on: [pull_request]
```

Comment:
Is there a way to run this only when a PR gets merged? IIUC, this will create a PR to the branch that modified the dependencies. I think what we want is to create a new PR to main once something that modifies the dependencies gets merged. This way we know that the commit being tagged is a commit that has been merged to main.

Reply:
interesting suggestion, I will look into that, thanks!

Reply:
The reason why we wanted a PR to PR was because e2e oftentimes fails if you don't retag the dependencies.

For this reason, we might either want a PR to PR or a direct commit to PR with the changes applied